compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/intrinsics/CompareTo.kt

```diff
@@ -84,6 +86,29 @@ class BooleanComparison(val op: IElementType, val a: MaterialValue, val b: Mater
     }
 }
 
+
+class NonIEEE754FloatComparison(val op: IElementType, val a: MaterialValue, val b: MaterialValue) : BooleanValue(a.codegen) {
+    private val numberCompareOpcode = NumberCompare.getNumberCompareOpcode(op)
+
+    override fun jumpIfFalse(target: Label) {
+        when (a.type) {
+            Type.FLOAT_TYPE -> mv.invokestatic("java/lang/Float", "compare", "(FF)I", false)
+            Type.DOUBLE_TYPE -> mv.invokestatic("java/lang/Double", "compare", "(DD)I", false)
+            else -> throw UnsupportedOperationException()
+        }
+        mv.visitJumpInsn(numberCompareOpcode, target)
+    }
+
+    override fun jumpIfTrue(target: Label) {
+        when (a.type) {
```

Comment:
Please extract the common part with `jumpIfFalse`

Reply:
Done, thanks!